src/metakb/normalizers.py

```diff
@@ -331,6 +218,41 @@ def get_regulatory_approval_extension(
 
         return regulatory_approval_extension
 
+    @staticmethod
+    def _normalize_concept(
+        query: str,
+        query_handler: GeneQueryHandler | DiseaseQueryHandler | TherapyQueryHandler,
+        concept_name: str,
+    ) -> tuple[NormalizedGene | NormalizedDisease | NormalizedTherapy, str | None]:
+        """Attempt to normalize a concept
+
+        :param query: Query to normalize
+        :param query_handler: Query handler for normalizer
+        :param concept_name: Name of concept (gene, disease, therapy)
+        :raises TokenRetrievalError: If AWS credentials are expired
+        :return: Normalizer response and normalized ID, if available.
+        """
+        normalizer_resp = None
+        normalized_id = None
+
+        try:
+            normalizer_resp = query_handler.normalize(query)
+        except TokenRetrievalError as e:
+            _logger.error(e)
+            raise e
+        except Exception as e:
+            _logger.error(
+                "%s Normalizer raised an exception using query %s: %s",
+                concept_name.capitalize(),
+                query,
+                e,
+            )
+        else:
+            if normalizer_resp.match_type:
+                normalized_id = getattr(normalizer_resp, concept_name).primaryCode.root
```

Comment:
naive question for my own learning, is this just a hanging `else?` doesn't this normally follow an `if` statement, how does this work?

Reply:
So for try/except blocks, `else` is executed if no exceptions were raised in the `try` block. More info here: https://www.geeksforgeeks.org/try-except-else-and-finally-in-python/